clients/client-shell/cmds/d2g/d2g.go

```diff
@@ -37,8 +38,14 @@ To convert a task definition (JSON), you must use the task definition flag (-t,
 func convert(cmd *cobra.Command, args []string) (err error) {
 	isTaskDef, _ := cmd.Flags().GetBool("task-def")
 	filePath, _ := cmd.Flags().GetString("file")
+	// Default file extension is json(temporary, handled for piped input)
+	fileExtension := "json"
 
-	input, err := userInput(filePath)
+	// Read file extension(temporary, handled for piped input)
+	if len(filePath) >= 4 {
+		fileExtension = filePath[len(filePath)-4:]
+	}
+	input, err := userInput(filePath, fileExtension)
```

Comment:
Rather than checking the file extension, it would be better to check the content. There are many variations for the file extension, including, but not limited to no extension, `.yaml`, `.yml`, `.YAML`, `.YML`, `.json`, `.JSON`. What matters really is only whether the content can be successfully interpreted as a particular format, and if it passes the payload validation (i.e. includes required fields, etc). For this, it is only necessary to attempt to parse the content in a particular format, and then decide based on whether the data was successfully parsed, whether the data is of the particular format you are trying.